spikes/data/dataservice/app/DataAccessLayer/dalActors.cs

```diff
@@ -77,37 +72,20 @@ public async Task<IEnumerable<Actor>> GetActorsAsync(ActorQueryParameters actorQ
                 return ac;
             }
 
-            string sql = ActorSelect;
-
-            int offset = actorQueryParameters.GetOffset();
-            int limit = actorQueryParameters.PageSize;
-
-            string offsetLimit = string.Format(CultureInfo.InvariantCulture, ActorOffset, offset, limit);
+            string sql = App.SearchService.GetActorIds(actorQueryParameters);
+            List<Actor> res;
 
-            if (!string.IsNullOrEmpty(actorQueryParameters.Q))
+            if (!string.IsNullOrEmpty(sql))
             {
-                // convert to lower and escape embedded '
-                actorQueryParameters.Q = actorQueryParameters.Q.Trim().ToLowerInvariant().Replace("'", "''", System.StringComparison.OrdinalIgnoreCase);
-
-                if (!string.IsNullOrEmpty(actorQueryParameters.Q))
-                {
-                    // get actors by a "like" search on name
-                    sql += string.Format(CultureInfo.InvariantCulture, $" and contains(m.textSearch, @q) ");
-                }
+                res = (List<Actor>)await InternalCosmosDBSqlQuery<Actor>(sql).ConfigureAwait(false);
             }
-
-            sql += ActorOrderBy + offsetLimit;
-
-            QueryDefinition queryDefinition = new QueryDefinition(sql);
-
-            if (!string.IsNullOrEmpty(actorQueryParameters.Q))
+            else
             {
-                queryDefinition.WithParameter("@q", actorQueryParameters.Q);
+                res = new List<Actor>();
             }
```

Comment:
I often do this at the top with declaration (line 76) and then I don't need the `else` block, because it is ready to be returned.  The upside, no `else` needed.  The downside, construction at the top that might not be necessary.  I'm not sure which is better in practice.